abracadabra/src/writer/hdf5_writer.hh

```diff
@@ -0,0 +1,39 @@
+#ifndef HDF5WRITER_H
+#define HDF5WRITER_H
+
+#include "hdf5_functions.hh"
+#include "g4-mandatory/trajectory.hh"
+
+#include <hdf5.h>
+#include <iostream>
+
+
+class hdf5_writer {
+public:
+    hdf5_writer();
+    ~hdf5_writer() {}
+
+    void open(std::string filename);
+    void close();
+
+    void write_run_info(const char* param_key, const char* param_value);
+    void write_particle_info(int evt_id, trajectory * trj, bool primary, int mother_id, float kin_energy);
+
+private:
+    size_t file_;
```

Comment:
In `open` this is bound to the result of `HF5create`, which does *not* return `size_t`!

Similar comments apply to other members in this class.

Reply:
Fixed in all members.